File: frontend/src/api/staff.ts
import api from "./client";
import type { ResultadoVisita } from "./qr";
import type { Cupon } from "./cupones";
import type { Canje } from "./canjes";

export interface ClienteInfoStaff {
  id: string;
  nombre: string;
  email: string | null;
  whatsapp: string | null;
  codigoCliente: string;
}

export interface RelacionInfoStaff {
  visitasTotales: number;
  rachaActual: number;
  puntos: number;
  segmento: string;
}

export interface ClienteStaffResponse {
  cliente: ClienteInfoStaff;
  relacion: RelacionInfoStaff;
  cuponesDisponibles: Cupon[];
  canjesRecientes: Canje[];
}

export interface VisitaStaffResponse {
  clienteNombre: string;
  resultado: ResultadoVisita;
}

export interface CanjeStaffResponse {
  clienteNombre: string;
  canje: Canje;
  resultado: ResultadoVisita;
}

export const staffApi = {
  clientePorCodigo: (codigoCliente: string) =>
    api.get<ClienteStaffResponse>(`/staff/cliente/${encodeURIComponent(codigoCliente)}`),

  /** El QR personal del cliente (/wallet/mi-qr) codifica welve://cliente/{id},
   * no su codigo_cliente WLV-XXXX — esta es la variante que reconoce eso. */
  clientePorQR: (clienteId: string) =>
    api.get<ClienteStaffResponse>(`/staff/cliente/por-qr/${encodeURIComponent(clienteId)}`),

  visitaPorCodigo: (codigoCliente: string, monto?: number) =>
    api.post<VisitaStaffResponse>("/staff/visita/por-codigo", { codigo_cliente: codigoCliente, monto }),

  visitaPorQR: (clienteId: string, monto?: number) =>
    api.post<VisitaStaffResponse>("/staff/visita/por-qr", { cliente_id: clienteId, monto }),

  canjePorCodigo: (codigoCliente: string, cuponId: string, monto?: number) =>
    api.post<CanjeStaffResponse>("/staff/canje/por-codigo", { codigo_cliente: codigoCliente, cupon_id: cuponId, monto }),

  canjePorQR: (clienteId: string, cuponId: string, monto?: number) =>
    api.post<CanjeStaffResponse>("/staff/canje/por-qr", { cliente_id: clienteId, cupon_id: cuponId, monto }),
};

File: backend/app/schemas/staff.py
from pydantic import BaseModel

from app.schemas.canje import CanjeResponse
from app.schemas.cupon import CuponResponse
from app.schemas.qr import ResultadoVisitaResponse


class VisitaPorCodigoRequest(BaseModel):
    codigo_cliente: str
    monto: float | None = None


class VisitaPorQRRequest(BaseModel):
    cliente_id: str
    monto: float | None = None


class CanjePorCodigoRequest(BaseModel):
    codigo_cliente: str
    cupon_id: str
    monto: float | None = None


class CanjePorQRRequest(BaseModel):
    cliente_id: str
    cupon_id: str
    monto: float | None = None


class ClienteInfoStaff(BaseModel):
    nombre: str
    email: str | None = None
    whatsapp: str | None = None
    codigoCliente: str


class RelacionInfoStaff(BaseModel):
    visitasTotales: int
    rachaActual: int
    puntos: int
    segmento: str


class ClienteStaffResponse(BaseModel):
    cliente: ClienteInfoStaff
    relacion: RelacionInfoStaff
    cuponesDisponibles: list[CuponResponse]
    canjesRecientes: list[CanjeResponse]


class VisitaStaffResponse(BaseModel):
    clienteNombre: str
    resultado: ResultadoVisitaResponse


class CanjeStaffResponse(BaseModel):
    clienteNombre: str
    canje: CanjeResponse
    resultado: ResultadoVisitaResponse
